cpp/src/Ice/Reference.cpp

```diff
@@ -1583,30 +1570,34 @@ IceInternal::RoutableReference::createConnection(const vector<EndpointIPtr>& all
                 {
                     connection->setAdapter(_routerInfo->getAdapter());
                 }
-                _callback->setConnection(connection, compress);
+                _response(connection, compress);
             }
 
             void setException(std::exception_ptr ex) final
             {
-                _callback->setException(ex);
+                _exception(ex);
             }
 
-            CB1(const RouterInfoPtr& routerInfo, const GetConnectionCallbackPtr& callback) :
-                _routerInfo(routerInfo), _callback(callback)
+            CB1(
```

Comment:
OutgoingConnectionFactory::CreateConnectionCallback is the next callback to remove. Its removal depends on #1778.